crates/client/devnet/src/lib.rs

```diff
@@ -628,4 +637,133 @@ mod tests {
+                            })
+                            .flatten()
+                            .collect(),
+                        signature: vec![], // Signature is filled in by `sign_and_add_invoke_tx`.
+                        nonce: nonce.into(),
+                        resource_bounds: ResourceBoundsMapping {
+                            l1_gas: ResourceBounds { max_amount: 60000, max_price_per_unit: 10000 },
+                            l2_gas: ResourceBounds { max_amount: 60000, max_price_per_unit: 10000 },
+                        },
+                        tip: 0,
+                        paymaster_data: vec![],
+                        account_deployment_data: vec![],
+                        nonce_data_availability_mode: starknet_core::types::DataAvailabilityMode::L1,
+                        fee_data_availability_mode: starknet_core::types::DataAvailabilityMode::L1,
+                        is_query: false,
+                    }),
+                    contract_0,
+                )
+                .unwrap();
+        }
+
+        let result = chain.sign_and_add_invoke_tx(
+            BroadcastedInvokeTransaction::V3(BroadcastedInvokeTransactionV3 {
+                sender_address: contract_0.address,
+                calldata: Multicall::default()
+                    .with(Call {
+                        to: ERC20_STRK_CONTRACT_ADDRESS,
+                        selector: Selector::from("transfer"),
+                        calldata: vec![contract_1.address, 15.into(), Felt::ZERO],
+                    })
+                    .flatten()
+                    .collect(),
+                signature: vec![], // Signature is filled in by `sign_and_add_invoke_tx`.
+                nonce: 5.into(),
+                resource_bounds: ResourceBoundsMapping {
+                    l1_gas: ResourceBounds { max_amount: 60000, max_price_per_unit: 10000 },
+                    l2_gas: ResourceBounds { max_amount: 60000, max_price_per_unit: 10000 },
+                },
+                tip: 0,
+                paymaster_data: vec![],
+                account_deployment_data: vec![],
+                nonce_data_availability_mode: starknet_core::types::DataAvailabilityMode::L1,
+                fee_data_availability_mode: starknet_core::types::DataAvailabilityMode::L1,
+                is_query: false,
+            }),
+            contract_0,
+        );
+
+        assert_matches!(
+            result,
+            Err(mc_mempool::Error::InnerMempool(mc_mempool::TxInsersionError::Limit(
+                mc_mempool::MempoolLimitReached::MaxTransactions { max: 5 }
+            )))
+        )
+    }
+
+    #[rstest]
+    fn test_mempool_age_limit() {
+        let max_age = Duration::from_millis(1000);
+        let mut chain =
+            chain_with_mempool_limits(MempoolLimits { max_age, max_declare_transactions: 2, max_transactions: 5 });
+        println!("{}", chain.contracts);
```

Comment:
this was intended actually, this is the println! that writes the devnet contracts keys to the console with the proper formatting and everything

i thought it made more sense to put it in stdout instead of inside a normal log, so that it can be parsed by other programs easily or whatnot
i'm fine with changing it but i was thinking that this made more sense that way

(this is not a change by the way, it was like that since the day i made the devnet)